### ISSUE TITLE
pip install sopel fails with sopel 6.6.0 and python 3.7 in fresh virtualenv

### ISSUE THREAD

**Issue by Corwind** [sopel-irc/sopel#1452]
OS: Arch Linux
Python: 3.7.2
Sopel: 6.6.0

On an up-to-date Arch Linux, with python 3.7.2, in a fresh virtualenv the installation process of sopel fails because of one of its dependencies: pyenchant which is unmaintained since the 24th of february of 2018.
This dependency fails to install, leading to sopel to fail to install.

```
$ pip install sopel
Collecting sopel
  Using cached https://files.pythonhosted.org/packages/3b/cf/397ac0b33c8cccc6877acb434c96692eeb853333bacceff36bb0613bb05e/sopel-6.6.0.tar.gz
Collecting xmltodict (from sopel)
  Using cached https://files.pythonhosted.org/packages/42/a9/7e99652c6bc619d19d58cdd8c47560730eb5825d43a7e25db2e1d776ceb7/xmltodict-0.11.0-py2.py3-none-any.whl
Collecting pytz (from sopel)
  Using cached https://files.pythonhosted.org/packages/61/28/1d3920e4d1d50b19bc5d24398a7cd85cc7b9a75a490570d5a30c57622d34/pytz-2018.9-py2.py3-none-any.whl
Collecting praw<6.0.0 (from sopel)
  Using cached https://files.pythonhosted.org/packages/95/71/f3a208d68bf93709c52b4fc110a3c8aab2f68215dcd171ee42c7c3671645/praw-5.4.0-py2.py3-none-any.whl
Collecting pyenchant (from sopel)
  Using cached https://files.pythonhosted.org/packages/9e/54/04d88a59efa33fefb88133ceb638cdf754319030c28aadc5a379d82140ed/pyenchant-2.0.0.tar.gz
    Complete output from command python setup.py egg_info:
    Traceback (most recent call last):
      File "<string>", line 1, in <module>
      File "/tmp/pip-install-kdnf9ejs/pyenchant/setup.py", line 212, in <module>
        import enchant
      File "/tmp/pip-install-kdnf9ejs/pyenchant/enchant/__init__.py", line 92, in <module>
        from enchant import _enchant as _e
      File "/tmp/pip-install-kdnf9ejs/pyenchant/enchant/_enchant.py", line 145, in <module>
        raise ImportError(msg)
    ImportError: The 'enchant' C library was not found. Please install it via your OS package manager, or use a pre-built binary wheel from PyPI.

    ----------------------------------------
Command "python setup.py egg_info" failed with error code 1 in /tmp/pip-install-kdnf9ejs/pyenchant/

```

The enchant library is installed on my system and I can install pyenchant from the Arch Linux package repositories, but I do not want to enable system packages to be used in this virtualenv.

This issue is related to #1422 and can be fixed by either deprecating the spellcheck module / removing pyenchant from the requirements.txt file or by merging #1164 and not using pyenchant anymore.

**Comment by dgw** [comment]
> This issue is related to #1422 and can be fixed by […] merging #1164 and not using pyenchant anymore.

Removing pyenchant has been planned for months. Since you found #1164, you also must have seen that it's scheduled for the next major release of Sopel, **7.0.0**.

The same error occurred the last time I tried to test-install Sopel on 64-bit Windows, even with a much older Python version. Between that and the py3.7 issues, I completely agree that `enchant` needs to be replaced. However, please understand that _this_ project was essentially unmaintained for about two years and we're still playing catch-up with all the changes that happened during that time. Trying to include every possible fix in version 6.6.0 would have delayed its release until _next_ January. :smile_cat:

Until Sopel 7 comes out with the rewritten spellcheck module, you can work around this by installing Sopel from source inside your virtualenv using a customized requirements.txt file.

**Comment by dgw** [comment]
Will release a version **6.6.1** shortly including this fix. It's going to be some time until Sopel 7 is ready, and there's no reason to force the use of pip + git installs until then.
